aptos-move/e2e-move-tests/src/harness.rs

```diff
@@ -680,40 +680,92 @@ impl MoveHarness {
 /// Helper to assert transaction is successful
 #[macro_export]
 macro_rules! assert_success {
-    ($s:expr) => {{
-        use aptos_types::transaction::*;
-        assert_eq!($s, TransactionStatus::Keep(ExecutionStatus::Success))
+    ($s:expr $(,)?) => {{
+        assert_eq!($s, aptos_types::transaction::TransactionStatus::Keep(
+            aptos_types::transaction::ExecutionStatus::Success))
+    }};
+    ($s:expr, $($arg:tt)+) => {{
+        assert_eq!(
+            $s,
+            aptos_types::transaction::TransactionStatus::Keep(
+                aptos_types::transaction::ExecutionStatus::Success),
+            $($arg)+,
+        )
     }};
 }
 
 /// Helper to assert transaction aborts.
 #[macro_export]
 macro_rules! assert_abort {
-    ($s:expr, $c:pat) => {{
+    // identity needs to be before pattern (both with and without message),
+    // as if we pass variable - it matches the pattern arm, but value is not used, but overriden.
+    // Opposite order and test_asserts_variable_used / test_asserts_variable_used_with_message tests
+    // would fail
+    ($s:expr, $c:ident $(,)?) => {{
+        assert!(matches!(
+            $s,
+            aptos_types::transaction::TransactionStatus::Keep(
+                aptos_types::transaction::ExecutionStatus::MoveAbort { code, .. }
+            )
+            if code == $c,
+        ));
+    }};
+    ($s:expr, $c:pat $(,)?) => {{
         assert!(matches!(
             $s,
             aptos_types::transaction::TransactionStatus::Keep(
                 aptos_types::transaction::ExecutionStatus::MoveAbort { code: $c, .. }
             ),
         ));
     }};
+    ($s:expr, $c:ident, $($arg:tt)+) => {{
+        assert!(
```

Comment:
nit: I think you can use `assert_matches!` directly

Reply:
not all can (this one cannot), so leaving for consistency